src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * This component is a menu for editing a link in the text editor.
+ * It allows the user to input the text and url for the link.
+ */
+@Component({
+    tag: 'limel-text-editor-link-menu',
+    shadow: true,
+    styleUrl: 'limel-text-editor-link-menu.scss',
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop({ reflect: true })
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop({ reflect: true })
+    public isOpen: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
+
+    @State()
+    private invalidLink: boolean = false;
+
+    public render() {
+        return [
+            <limel-input-field
+                label={this.getTranslation('editor-link-menu.text')}
+                value={this.link?.text || ''}
+                leadingIcon="text_cursor"
+                onChange={this.handleLinkTitleChange}
+            />,
+            <limel-input-field
+                label={this.getTranslation('editor-link-menu.link')}
+                value={this.link?.href || ''}
+                leadingIcon="-lime-text-link"
+                invalid={this.invalidLink}
+                onChange={this.handleLinkValueChange}
+            />,
+            <div class="actions">
+                <limel-button
+                    label={this.getTranslation('cancel')}
+                    onClick={this.handleCancel}
+                    slot="button"
+                />
+                <limel-button
+                    primary={true}
+                    label={this.getTranslation('save')}
+                    disabled={!this.link?.href || this.invalidLink}
+                    onClick={this.handleSave}
+                    slot="button"
+                />
+            </div>,
+        ];
+    }
+
+    private getTranslation = (key: string) => {
+        return translate.get(key, this.language);
+    };
+
+    private validateLink = (href: string): boolean => {
+        if (!hasKnownProtocol(href)) {
+            const formattedLink = prependProtocol(href);
+
+            return !!getHref(formattedLink);
+        }
+
+        return !!getHref(href);
+    };
+
+    private handleCancel = () => {
+        this.cancel.emit();
+    };
+
+    private handleSave = () => {
+        this.save.emit();
```

Comment:
I wonder if we should stop the propagation for the events, since we have had trouble in the past with multiple events flying around under the same name

Reply:
good catch!

Reply:
⚡